modules/foreman_global_parameter.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c)Matthias Dellweg & Bernhard Hopfenmüller 2017
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: foreman_global_parameter
+short_description: Manage Foreman Global Parameters
+description:
+    - "Manage Foreman Global Parameter Entities"
+    - "Uses https://github.com/SatelliteQE/nailgun"
+version_added: "2.4"
+author:
+- "Bernhard Hopfenmueller(@Fobhep)"
+- "Matthias Dellweg (@mdellweg)"
+requirements:
+    - nailgun >= 0.29.0
+options:
+    server_url:
+        description:
+        - URL of Foreman server
+        required: true
+    username:
+        description:
+        - Username on Foreman server
+        required: true
+        default: true
+    password:
+        description:
+        - Password for user accessing Foreman server
+        required: true
+    verify_ssl:
+        description:
+        - Verify SSL of the Foreman server
+        required: false
+    name:
+        description:
+        - Name of the Global Parameter
+        required: true
+    value:
+        description:
+        - Value of the Global Parameter
+        required: false
+    state:
+        description:
+        - State of the Global Parameter
+        required: false
+        choices:
+        - present
+        - absent
+'''
+
+EXAMPLES = '''
+- name: "Create a Global Parameter"
+  local_action:
+      module: foreman_global_parameter
+      username: "admin"
+      password: "admin"
+      server_url: "https://fakeserver.com"
+      name: "TheAnswer"
+      value: "42"
+      state: present
+
+- name: "Delete a Global Parameter"
+  local_action:
+      module: foreman_global_parameter
+      username: "admin"
+      password: "admin"
+      server_url: "https://fakeserver.com"
+      name: "TheAnswer"
+      state: absent
+'''
```

Comment:
Do you mind updating the examples to follow the updated way of declaring them such as https://github.com/theforeman/foreman-ansible-modules/blob/master/modules/katello_publish.py#L54 ? This reduces amount of declared things and makes the module top level.